mod/primitives/pkg/chain/helpers_test.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2024, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package chain
+
+import (
+	"testing"
+
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/version"
+	"github.com/stretchr/testify/assert"
+)
+
+// Define concrete types for the generic parameters
+type domainTypeT [4]byte
+type epochT uint64
+type executionAddressT [20]byte
+type slotT uint64
+type cometBFTConfigT struct{}
+
+// Create an instance of chainSpec with test data
+var spec = chainSpec[domainTypeT, epochT, executionAddressT, slotT, cometBFTConfigT] {
+	Data: SpecData[
+		domainTypeT, epochT, executionAddressT, slotT, cometBFTConfigT,
+	]{
+		ElectraForkEpoch: 10,
+		SlotsPerEpoch: 32,
+		MinEpochsForBlobsSidecarsRequest: 5,
+	},
+}
+
+// TestActiveForkVersionForEpoch tests the ActiveForkVersionForEpoch method
+func TestActiveForkVersionForEpoch(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		epoch    epochT
+		expected uint32
+	}{
+		{epoch: 9, expected: version.Deneb},
+		{epoch: 10, expected: version.Electra},
+		{epoch: 11, expected: version.Electra},
+	}
+
+	// Run test cases
+	for _, tc := range testCases {
+		result := spec.ActiveForkVersionForEpoch(tc.epoch)
+		assert.Equal(t, tc.expected, result, "unexpected active fork version for epoch %d", tc.epoch)
```

Comment:
Q: Do we want the tests to be executed after assertion failure ? Was wondering should we use assert or require.

Reply:
hi @nidhi-singh02, imo using assert is suitable for our test cases because each test case is independent, and we want to see the results for all the defined epochs

Reply:
we should use require not assert ya?

Reply:
@nidhi-singh02 @itsdevbear thanks guys, I just refactored it so please help me review it again